source/x509-ldap.rst

```diff
@@ -0,0 +1,239 @@
+   .. code-block:: bash
+
+      $ cat server.key server.crt > server.pem
+
+Generate client certificates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Generate client certificate request and key. In the ``-subj`` flag, specify the information about clients in the :abbr:`DN (Distinguished Name)` format.
+   
+   .. code-block:: bash
+
+      $ openssl req -nodes -newkey rsa:4096 -keyout client.key -out client.csr -subj "/DC=com/DC=percona/CN=John Doe"
+
+#. Sign the client certificate request with the root CA certificate.
+   
+   .. code-block:: bash
+
+      $ openssl x509 -req -in client.csr -CA ca.crt -CAkey ca.key -set_serial 02 -out client.crt
+
+#. Combine the client certificate and key to create a certificate key file.
+   
+   .. code-block:: bash
+
+      $ cat client.key client.crt > client.pem
+
+Set up the LDAP server
+-------------------------
+
+The setup of the LDAP server and the configuration of the LDAP schema is out of scope of this document. We assume that you have LDAP server up and running and accessible to |PSMDB|.
+
+Configure ``mongod`` server
+-----------------------------
+
+The configuration consists of the following steps:
+
+* Creating a role that matches the user group on the LDAP server
+* Editing the configuration file to enable the x.509 authentication
+
+Create roles 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+At this step, create the roles in the ``admin`` database with the names that exactly match the names of the user groups on the LDAP server. These roles are used for user :ref:`authorization` in |PSMDB|. 
+
+In our example, we create the role `cn=otherusers,dc=percona,dc=com` that has the corresponding LDAP group.
+
+.. code-block:: javascript
+
+   var admin = db.getSiblingDB("admin")
+   db.createRole(
+      {
+        role: "cn=otherusers,dc=percona,dc=com", 
+        privileges: [], 
+        roles: [ 
+              "userAdminAnyDatabase", 
+              "clusterMonitor", 
+              "clusterManager", 
+              "clusterAdmin"
+               ]
+      }
+   )
+
+
+Output:
+
+.. code-block:: json
+
+   {
+   	"role" : "cn=otherusers,dc=percona,dc=com",
+   	"privileges" : [ ],
+   	"roles" : [
+   		"userAdminAnyDatabase",
+   		"clusterMonitor",
+   		"clusterManager",
+   		"clusterAdmin"
+   	]
+   }
+
+
+Enable x.509 authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Stop the ``mongod`` service
+   
+   .. code-block:: bash
+
+      $ sudo systemctl stop mongod
+
+#. Edit the :file:`/etc/mongod.conf` configuration file.
+   
+   .. code-block:: yaml
+
+      net:
+        port: 27017
+        bindIp: 127.0.0.1
+        tls:
+         mode: requireTLS
+         certificateKeyFile: /var/lib/mongocerts/server.pem
+         CAFile: /var/lib/mongocerts/ca.crt
+
+      security:
+        authorization: enabled
+        ldap:
+         servers: "ldap.example.com"
+         transportSecurity: none
+         authz:
+          queryTemplate: "dc=percona,dc=com??sub?(&(objectClass=groupOfNames)(member={USER}))"
```

Comment:
inconsistent indentation. I would recommend to use 2 spaces everywhere

Reply:
Changed